Mathlib/Tactic/Eval.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+
+import Qq.Macro
+import Std.Lean.Expr
+
+/-!
+# The `eval%` term elaborator
+
+This file provides the `eval% x` term elaborator, which evaluates the constant `x` at compile-time
+in the interpreter, and interpolates it into the expression.
+-/
+
+
+open Qq Lean Elab Term
+
+namespace Mathlib.Meta
+
+/--
+`eval% x` evaluates the term `x : X` in the interpreter, and then injects the resulting expression.
+
+As an example:
+```lean
+example : 2^10 = eval% 2^10 := by
+  -- goal is `2^10 = 1024`
+  sorry
+```
+This only works if a `Lean.ToExpr X` instance is available.
+-/
```

Comment:
```suggestion

Using 'show_term eval% x' displays what is the value of 'eval% x'.
-/
```
Do you think that this might be useful?

Reply:
How about phrased like this?

```suggestion

Tip: you can use `show_term eval% x` to get the value of `eval% x`.
-/
```

Let's go ahead with including `eval%`:

bors d+

Reply:
Added, but I replaced `get` with `see` to avoid any confusion about how to store the gotten value.